### ISSUE TITLE
[BUG] [Java/Spring] Inline ENUMs not generated and resolved as plain type

### ISSUE THREAD

**Issue by anthochristen** [OpenAPITools/openapi-generator#17992]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [ ] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

Enums defined inline in the parameters section of a spec is resolved as primitive type and not an enum.

##### openapi-generator version

6.6.0 , 7.3.0

##### OpenAPI declaration file content or url

```json
        "parameters": [
          {
            "name": "inline-enum",
            "description": "There can be only some!",
            "in": "query",
            "schema": {
              "type": "array",
              "items": {
                "type": "integer",
                "format": "int32",
                "enum": [
                  3,
                  4,
                  7,
                  9
                ],
                "default": 4
              }
            }
          }
        ]
```
[GIST link](https://gist.github.com/anthochristen/cdce784c37f1169dc80e5ad5189cdba3)

##### Generation Details

```bash
 java -jar openapi-generator-cli-7.3.0.jar generate -i openapi.json -g spring --library spring-boot -o ./tes
```

**Comment by jpfinne** [comment]
Your issue is only for parameters.
The inline enum works for models.

Where do want the enum?   inside NoSecureStuffApi.java as `public enum ListEnum {}`?
IHMO not as convenient as a standalone class with a well defined class name.

The parsing of parameters (CodeGenParameter) is not as complete as the parsing of models (CodeGenModel).
For example a query param using an exploded object does not support allOf.

**Comment by anthochristen** [comment]
Yes, its only for parameters. Having enums inside Resource classes might not be the best idea. But when [enum example 101](https://swagger.io/docs/specification/data-models/enums/) on swagger.io doesn't work it feels wrong.

Perhaps they should be made standalone enum model classes with path suffix?

**Comment by matej-staron** [comment]
Duplicate of #13493
